api_app/core/serializers.py

```diff
@@ -163,6 +165,31 @@ def _md5_config_file(cls) -> str:
             md5hash = hashlib.md5(buffer).hexdigest()
         return md5hash
 
+    @classmethod
+    def _complete_config(
+        cls, config_dict: dict, plugin_name: str, visited: set
+    ) -> dict:
+        """
+        Completes config by parsing extends configs
+        """
+        if plugin_name in visited:
+            raise RuntimeError(f"Circular dependency detected in {cls} config")
+        visited.add(plugin_name)
+        if plugin_name not in config_dict:
+            raise RuntimeError(
+                f"Plugin {plugin_name} not found in {cls} config "
+                "but referenced in extends"
+            )
+        if "extends" in config_dict[plugin_name]:
+            temp = config_dict[plugin_name]
+            parent_plugin = temp.pop("extends")
+            config_dict[plugin_name] = deepcopy(
+                cls._complete_config(config_dict, parent_plugin, visited)
+            )
+            for key in temp:
+                config_dict[plugin_name][key] = temp[key]
+        return config_dict[plugin_name]
```

Comment:
Fun of recursion, but not a fan of modifying inputs. I would have suggest that you take the input, deepcopy it and return the new dict as output. Then, where you can write  
```python3
        for key, plugin in config_dict.items():
            config_dict[key] = cls._complete_config(config_dict, plugin, set())
```
Now, i see that this is technically slower, because you are making _another_ copy of the dictionary, but Imho the readability/maintainability > 0.2% increased speed.